Scripts/Mac/BuildGVFSForMac.sh

```diff
@@ -50,7 +50,7 @@ $VFS_SCRIPTDIR/GenerateCommonAssemblyVersion.sh $VERSION || exit 1
 # overriding build targets. The duplicate import generates warning MSB4011 that is ignored
 # by this switch.
 echo 'Restoring packages...'
-dotnet restore $VFS_SRCDIR/GVFS.sln /p:Configuration=$CONFIGURATION.Mac --packages $VFS_PACKAGESDIR /warnasmessage:MSB4011 || exit 1
+dotnet restore $VFS_SRCDIR/GVFS.sln --runtime osx-x64 /p:Configuration=$CONFIGURATION.Mac --packages $VFS_PACKAGESDIR /warnasmessage:MSB4011 || exit 1
```

Comment:
>My understanding is that this can happen when the runtime used by the restore
operation does not match the runtime expected by the publish operation. As the
publish operation might not call restore itself, it will generate this error
when the runtime does not match the expected version.

The change looks simple enough, but I am having trouble understanding how passing in `--runtime osx-x64` makes a difference.  Does this mean that previously some step in restore or publish was not using osx-x64?

Reply:
I think this means previously restore was not explicity using osx-64 from the botnet command, and instead would have picked it up from the project files. From the documentation:

> This is used to restore packages for runtimes not explicitly listed in the <RuntimeIdentifiers> tag in the .csproj file.

I am not sure if this change is absolutely necessary, as many (all?) of our projects specify the runtime identifier as well, but it seemed reasonable, as this is the only runtime we are targeting, and the other commands include it.

Reply:
Is this command even necessary at all still? With NETCore 2.0, the restore should be run implicitly by other dotnet commands...

https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-restore?tabs=netcore2x

Reply:
Correct.. however there are can be subtle differences when it comes to differing package reference styles, and if you're mixing old and new MSBuild project formats. Since this is for the _Mac_ build only (which should be .NET Core only, and MSBuild 15 only) it should be ok to drop the explicit `restore`.

Although it should still be tested!